test/traversals/diffusion.jl

```diff
@@ -0,0 +1,181 @@
+
+@testset "Diffusion Simulation" begin
+
+    ######
+    # Check on fully connected, prob = 1
+    ######
+
+    g = CompleteGraph(5)
+
+    # Most basic
+    result = @inferred( diffusion_rate(g, 1.0, 4) )
+    result
+    @test result == [1, 5, 5, 5]
```

Comment:
since you're not reusing `result`, it's more straightforward to do
```
@test @inferred(diffusion_rate(g, 1.0, 4)) == [1, 5, 5, 5]
```

That way you avoid the situation where you have an inference problem that causes an "Error outside test" – a failure of `@inferred` here creates a `TestError`.

Should probably change the rest of them as well.